javascript-challenges.test.js

```diff
@@ -38,6 +89,48 @@ const hand3 = [5, 5, 5, 5, 4]
 const hand4 = [7, 2, 7, 2, 7]
 // Expected output: true
 
+describe("fullHouse", () => {
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        expect(fullHouse(hand1)).toEqual(true)
+        expect(fullHouse(hand2)).toEqual(false)
+        expect(fullHouse(hand3)).toEqual(false)
+        expect(fullHouse(hand4)).toEqual(true)
+        
+    })
+})
+
 // b) Create the function that makes the test pass.
 
+
+
+function isFullHouse(hand) {
+    const uniqueNumbers = Array.from(new Set(hand));
+  
+    // Check for a pair and a three of a kind
+    let hasPair = false;
+    let hasThreeOfAKind = false;
+  
+    uniqueNumbers.forEach(number => {
+      const count = hand.filter(x => x === number).length;
+  
+      if (count === 2) {
+        hasPair = true;
+      } else if (count === 3) {
+        hasThreeOfAKind = true;
+      }
+    });
+  
+    // Return true if both conditions are met
+    return hasPair && hasThreeOfAKind;
+  }
+  
+  // Test cases
+  console.log(isFullHouse([5, 5, 5, 3, 3])); // true
+  console.log(isFullHouse([5, 5, 3, 3, 4])); // false
+  console.log(isFullHouse([5, 5, 5, 5, 4])); // false
+  console.log(isFullHouse([7, 2, 7, 2, 7])); // true
+  
+
+
 // Pseudo code:
+// I found the correct anwser of Chat . I just couldnt figure it out. Started getting frustrated and tired 
```

Comment:
Thank for your honesty. I wouldn't say this is the 'correct' answer. It's AN answer.  There are ways to do it with methods and concepts we've covered.  Give it another try.  I'll give you a hint:  ever play cards?  What's the first thing you do when you get your hand?